esi_leap/api/controllers/v1/offer.py

```diff
@@ -103,7 +104,17 @@ def post(self, new_offer):
         cdict = request.to_policy_values()
         policy.authorize('esi_leap:offer:create', cdict, cdict)
 
-        o = offer.Offer(**new_offer.to_dict())
+        offer_dict = new_offer.to_dict()
+
+        if 'project_id' in offer_dict:
+            if offer_dict['project_id'] != request.project_id:
+                policy.authorize('esi_leap:contract:offer_admin', cdict, cdict)
+        else:
+            offer_dict['project_id'] = request.project_id
+
+        OffersController.verify_resource_permission(cdict, offer_dict)
```

Comment:
Use `self.verify_resource_permission`

Reply:
Is there a reason this can't be a static method?

Reply:
No, you're right - ignore this.